### DISCUSSION TITLE
Porting to Pillow as recommended does not work for me

### DISCUSSION THREAD

**Discussion by ferdasonmez** [python-pillow/Pillow#7757] [Q&A] [answered]
I am using Python 3.11. I installed pillow 10.2.0.  I am getting error for this (and similar lines)` from PIL import Image`. The error is ModuleNotFoundError: No module named 'PIL'.

I am trying to use a software which I was able to use before. Please help. 

Regards,
Ferda

**Comment by nulano** [comment]
`ModuleNotFoundError: No module named 'PIL'`: This most likely means that Pillow is not installed.

How did you install Pillow?
How are you launching the software that uses it?

**Comment by ferdasonmez** [comment]
Hi, thanks for the quick response. I am using pycharm, I installed several times both using the pip install command and also via Pycharm package manager.

*Kind Regards,*
*Dr. Ferda Özdemir Sönmez*

**Reply by radarhere** [reply]
> using the pip install

What is the exact command you used?

**Reply by aclark4life** [reply]
Hi Ferda! 

Yeah, I think we'll need to see a screenshot or paste of your Pillow installation in order to provide any additional meaningful help. 

Pillow installation is complex, up to a point, after which the complexity lies in understanding the environment in which you are installing it. 🤔

**Comment by ferdasonmez** [comment]
I used pip install pillow I guess, later I tried to select specific versions. Currently, I switched back to python 2.7 for this specific project only to make it work. Still, not successful. This is what I can not try it again. What should I do?

*Kind Regards,*
*Dr. Ferda Özdemir Sönmez*

**Comment by radarhere** [comment]
Looking at what has helped other people install Pillow in PyCharm
- "I had to go to the Project Interpreter and install there the Pillow package" - https://github.com/python-pillow/Pillow/issues/4319#issuecomment-569822727, and also https://github.com/python-pillow/Pillow/issues/3117#issuecomment-513528512
- Upgrading pip - https://github.com/python-pillow/Pillow/issues/4530#issuecomment-610028058

**Comment by ferdasonmez** [comment]
Hello Andrew,
Of course, I am using the project interpreter and my pip is already up to
date. I think the problem is more complex than this.

*Kind Regards,*
*Dr. Ferda Özdemir Sönmez*

**Comment by ferdasonmez** [comment]
Hi,
Here is my screenshot. This shows all the details. The code is imported as
below ( I also tried other versions but none of them worked).

from PIL import Image, ImageDraw, ImageFont
from PIL.PngImagePlugin import PngImageFile, PngInfo

I am getting the error,
   from PIL import Image, ImageDraw, ImageFont
ModuleNotFoundError: No module named 'PIL'

I hope you can solve the problem because this took lots of time already
with no success :(

[image: image.png]
[image: image.png]
*Kind Regards,*
*Dr. Ferda Özdemir Sönmez*

**Reply by radarhere** [reply]
Unfortunately, your screenshots did not appear correctly.

**Comment by ferdasonmez** [comment]
What do you mean?

*Kind Regards,*
*Dr. Ferda Özdemir Sönmez*

**Reply by radarhere** [reply]
This is how your previous comment appeared. The images you attempted to include did not appear.

![Screenshot 2024-01-30 at 12 12 18 am](https://github.com/python-pillow/Pillow/assets/3112309/80708b91-8af8-4413-a57e-2a91d51defa8)

**Reply by ferdasonmez** [reply]
Hi,
Here is my screenshot. This shows all the details. The code is imported as below ( I also tried other versions but none of them worked).

from PIL import Image, ImageDraw, ImageFont
from PIL.PngImagePlugin import PngImageFile, PngInfo
![2](https://github.com/python-pillow/Pillow/assets/8194836/048d5bc0-24ed-4596-a7a1-be8f43908c86)

![3](https://github.com/python-pillow/Pillow/assets/8194836/fbc2a7db-1f70-4e9e-a3d1-45613ca32f89)

I am getting the error, 
   from PIL import Image, ImageDraw, ImageFont
ModuleNotFoundError: No module named 'PIL'

I hope you can solve the problem because this took lots of time already with no success :(



Kind Regards,
Dr. Ferda Özdemir Sönmez

**Reply by hugovk** [reply]
Before the Pillow import, can you try this:

```python
import sys
print(sys.version)
print(sys.executable)
```
And let us know what it says?

**Reply by ferdasonmez** [reply]
I put it to a new py file in the same folder as the other file(which does not compile).
the result is  3.11.3 (tags/v3.11.3:f3909b8, Apr  4 2023, 23:49:59) [MSC v.1934 64 bit (AMD64)]
C:\Programfiles\Python311\python.exe

**Reply by hugovk** [reply]
Does `from PIL import Image, ImageDraw, ImageFont` work in this new file?

Even if the old file causes a error, it would still be interesting to see what it prints out before the error.

**Comment by ferdasonmez** [comment]
Interestingly, yes.

*Kind Regards,*
*Dr. Ferda Özdemir Sönmez*

**Reply by hugovk** [reply]
Okay, so the new script is fine. 

Please can you add the prints to your original script, before the imports?

**Comment by ferdasonmez** [comment]
This is the same project, and same folder but I am getting a different result.

3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64 bit (AMD64)]
C:\Users\aax374\PycharmProjects\DAOProject1\venv\Scripts\python.exe
    from PIL import Image, ImageDraw, ImageFont
ModuleNotFoundError: No module named 'PIL'

**Reply by radarhere** [reply]
Ok, so this is running within a virtual environment.

Could you run `C:\Users\aax374\PycharmProjects\DAOProject1\venv\Scripts\python.exe -m pip install Pillow` from the command line, and then try running your script again?

**Comment by nulano** [comment]
If you'd like a PyCharm solution not involving the terminal, I imagine you have a different interpreter selected in your run configuration. See https://www.jetbrains.com/help/pycharm/run-debug-configuration.html#da5a5665 for editing run configurations - this is probably set to something other than "Project Default (...)".

If instead you want to use that interpreter for that configuration and just need to install Pillow in it, you can change the selected interpreter in the project settings, install Pillow using the PyCharm interface, then switch the interpreter back. See also https://www.jetbrains.com/help/pycharm/configuring-python-interpreter.html#modify_interpreter

**Reply by radarhere** [reply]
@ferdasonmez did either this or [my suggestion](https://github.com/python-pillow/Pillow/discussions/7757#discussioncomment-8292915) work?

**Comment by ferdasonmez** [comment, marked as answer]
Him, When I realized that one file running another python version, I
recreated it and now it is ok,. Thank you..


*Kind Regards,*
*Dr. Ferda Özdemir Sönmez*
